packages/syft/src/syft/ast/klass.py

```diff
@@ -531,6 +531,22 @@ def create_pointer_class(self) -> None:
         setattr(klass_pointer, "path_and_name", self.path_and_name)
         setattr(self, self.pointer_name, klass_pointer)
 
+    def store_init_args(outer_self: Any) -> None:
+        """
+        Stores args and kwargs of outer_self init by wrapping the init method.
+        """
+
+        def init_wrapper(self: Any, *args: List[Any], **kwargs: Dict[Any, Any]) -> None:
+            outer_self.object_ref._wrapped_init(self, *args, **kwargs)
+            self._init_args = args
+            self._init_kwargs = kwargs
+
+        # If _wrapped_init already exists, create_init_method is already called once
+        # and does not need to wrap __init__ again.
+        if not hasattr(outer_self.object_ref, "_wrapped_init"):
+            outer_self.object_ref._wrapped_init = outer_self.object_ref.__init__
+            outer_self.object_ref.__init__ = init_wrapper
+
```

Comment:
Could you explain why this was a problem? It might be a good thing to do, not exactly sure about the use case/if it works as you are expecting.